forch/radius_query.py

```diff
@@ -0,0 +1,113 @@
+"""Talks and listens to RADIUS. Takes a packet object as input"""
+from queue import Queue
+import socket
+import logging
+import os
+from time import sleep
+import json
+
+from forch.radius import RadiusAttributesList, RadiusAccessRequest, Radius
+from forch.radius_attributes import CallingStationId, UserName, MessageAuthenticator, \
+        NASPort, UserPassword
+from forch.utils import MessageParseError
+
+LOGGER = logging.getLogger('rquery')
+
+RADIUS_HEADER_LENGTH = 1 + 1 + 2 + 16
+
+class RadiusSocket:
+    """Handle the RADIUS socket"""
+
+    def __init__(self, listen_ip, listen_port, server_ip,  # pylint: disable=too-many-arguments
+                 server_port):
+        self.socket = None
+        self.listen_ip = listen_ip
+        self.listen_port = listen_port
+        self.server_ip = server_ip
+        self.server_port = server_port
+
+    def setup(self):
```

Comment:
seems like this code is duplicate with radius_socket.py?

Reply:
You are right. Let me correct that.

Reply:
Done.